account_payment_order/models/account_payment_mode.py

```diff
@@ -0,0 +1,148 @@
+        domain=[('reconcile', '=', True)],
+        help="Pay off lines in 'file uploaded' payment orders with a move on "
+        "this account. You can only select accounts "
+        "that are marked for reconciliation")
+    transfer_journal_id = fields.Many2one(
+        'account.journal', string='Transfer Journal',
+        help='Journal to write payment entries when confirming '
+        'payment/debit orders of this mode')
+    move_option = fields.Selection([
+        ('date', 'One move per payment date'),
+        ('line', 'One move per payment line'),
+        ], string='Move Option', default='date')
+    post_move = fields.Boolean(string='Post Move', default=True)
+
+    @api.multi
+    @api.constrains(
+        'generate_move', 'offsetting_account',
+        'transfer_account_id', 'transfer_journal_id', 'move_option')
+    def transfer_move_constrains(self):
+        for mode in self:
+            if mode.generate_move:
+                if not mode.offsetting_account:
+                    raise ValidationError(_(
+                        "On the payment mode '%s', you must select an "
+                        "option for the 'Offsetting Account' parameter")
+                        % mode.name)
+                elif mode.offsetting_account == 'transfer_account':
+                    if not mode.transfer_account_id:
+                        raise ValidationError(_(
+                            "On the payment mode '%s', you must "
+                            "select a value for the 'Transfer Account'.")
+                            % mode.name)
+                    if not mode.transfer_journal_id:
+                        raise ValidationError(_(
+                            "On the payment mode '%s', you must "
+                            "select a value for the 'Transfer Journal'.")
+                            % mode.name)
+                if not mode.move_option:
+                    raise ValidationError(_(
+                        "On the payment mode '%s', you must "
+                        "choose an option for the 'Move Option' "
+                        "parameter.") % mode.name)
+
+    @api.onchange('payment_method_id')
+    def payment_method_id_change(self):
+        if self.payment_method_id:
+            ajo = self.env['account.journal']
+            aj_ids = []
+            if self.payment_method_id.payment_type == 'outbound':
+                aj_ids = ajo.search([
+                    ('type', 'in', ('purchase_refund', 'purchase')),
+                    ('company_id', '=', self.company_id.id)
+                ]).ids
+            elif self.payment_method_id.payment_type == 'inbound':
+                aj_ids = ajo.search([
+                    ('type', 'in', ('sale_refund', 'sale')),
+                    ('company_id', '=', self.company_id.id)
+                ]).ids
+            self.default_journal_ids = [(6, 0, aj_ids)]
+
+    @api.onchange('generate_move')
+    def generate_move_change(self):
```

Comment:
It would not be better with @api.multi and write?

Reply:
An onchange mustn't write ever, as it's a on-memory operation. `@api.multi` is by default, but by convention this should be always applied to only one record.

Reply:
It's good know it. Thanks @pedrobaeza